### PR TITLE
refactor: replace tls lib v3 with v4 for access

### PR DESCRIPTION
**Pull request by gruyaume**
# Description

Replace the TLS certificates library V3 with V4 for the access integration. 

## Use of self-signed certificates
The Vault charm can effectively operate in two modes for its own tls certificates:
- Self Signed
- TLS integration

Here we explicitly state those two modes. When we use self signed we generate our own certs and when we use the tls integration, we ask certs using the integration.

## This change does not include the PKI integration
In order to make this change a bit smaller we only include here the Access integration. A separate PR will focus on the PKI integration.

# Checklist:

- [ ] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that validate the behaviour of the software
- [ ] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: lib/charms/vault_k8s/v0/vault_tls.py [hunk 1]
```diff
@@ -136,90 +150,152 @@ def __init__(
...
-            self.tls_access.on.certificate_invalidated, self._on_certificate_config_changed
+
+    # def _on_tls_certificates_access_relation_broken(self, event: RelationBrokenEvent):
+    #     """Handle leaving the tls access relation.
+
+    #     Regenerates self signed certificates from the saved self generated root CA
+    #     and reloads vault.
+    #     """
+    #     self._remove_all_certs_from_workload()
+    #     self.charm.on.config_changed.emit()
```
Comment (ghislainbourgeois) [inline lib/charms/vault_k8s/v0/vault_tls.py:205]: Is this still needed?

File: lib/charms/vault_k8s/v0/vault_tls.py [hunk 2]
```diff
@@ -402,23 +406,12 @@ def ca_certificate_is_saved(self) -> bool:
...
-            self.workload.send_signal(signal=SIGHUP, process=self._service_name)
-            logger.debug("Vault reload requested")
-        except Exception:
-            logger.debug("Couldn't send signal to process. Proceeding normally.")
+    # def _remove_all_certs_from_workload(self) -> None:
+    #     """Remove the certificate files that are used for authentication."""
+    #     self._remove_tls_file_from_workload(File.CA)
+    #     self._remove_tls_file_from_workload(File.CERT)
+    #     self._remove_tls_file_from_workload(File.KEY)
+    #     logger.debug("Removed existing certificate files from workload.")
```
Comment (ghislainbourgeois) [inline lib/charms/vault_k8s/v0/vault_tls.py:414]: Is this still required?

**Review by ghislainbourgeois — CHANGES_REQUESTED** [review]
Looks good, just 2 small commented out sections I think you forgot.

File: lib/charms/vault_k8s/v0/vault_tls.py (hunk 1 above)
Comment (gruyaume) [reply lib/charms/vault_k8s/v0/vault_tls.py]: Oops I should have delete this. Done.

File: lib/charms/vault_k8s/v0/vault_tls.py (hunk 2 above)
Comment (gruyaume) [reply lib/charms/vault_k8s/v0/vault_tls.py]: Oops I should have delete this. Done.

**Comment by kayra1** [comment]
@saltiyazan this is the same issue you were having with the integration tests. How did you resolve it?

**Review by ghislainbourgeois — APPROVED** [review]

**Pull request merged**